.github/workflows/libc-overlay-tests.yml

```diff
@@ -0,0 +1,85 @@
+# This workflow is for pre-commit testing of the LLVM-libc project.
+name: LLVM-libc Pre-commit Overlay Tests
+
+on:
+  pull_request:
+    branches: [ "main" ]
+    paths:
+      - 'libc/**'
+      - '.github/workflows/libc-overlay-tests.yml'
+
+jobs:
+  build:
+    env:
+      SCCACHE_GHA_ENABLED: "true"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      # Set fail-fast to false to ensure that feedback is delivered for all matrix combinations.
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-24.04, windows-2022, macos-14 ]
+        compiler: [
+          { c_compiler: gcc, cpp_compiler: g++ },
+          { c_compiler: clang, cpp_compiler: clang++ },
+          { c_compiler: clang-cl, cpp_compiler: clang-cl }
+        ]
+        exclude:
+          - os: windows-2022
+            compiler: { c_compiler: gcc, cpp_compiler: g++ }
+          - os: windows-2022
+            compiler: { c_compiler: clang, cpp_compiler: clang++ }
+          - os: ubuntu-24.04
+            compiler: { c_compiler: clang-cl, cpp_compiler: clang-cl }
+          - os: macos-14
+            compiler: { c_compiler: clang-cl, cpp_compiler: clang-cl }
+          - os: macos-14
+            compiler: { c_compiler: gcc, cpp_compiler: g++ }
```

Comment:
I think it might be easier and more readable to use `include` here instead of `exclude`, but it's up to you.